GitHub policies (#740)
digidem/mapeo-mobile: GitHubPolicy.md

# Github Policy and Best Practices
<!-- This is WIP, with some of the notes and thoughts not fully developed -->

<!-- Add links when completed -->
## Table of Contents 
1. [Overview](#overview)
2. [Glossary of Terms](#glossary-of-term)
3. [Best Practices](#best-practices)
4. [Giving Feeback](#giving-feeback)
5. [Github Pull Request Template](#template)
6. [How To:](#how-to)
    * [Create a new branch]()
    * [Change base branch]()
    * [Create a PR]()
    * [Merge]()
    * [Rebase]()
    * [Squash and Merge]()
    * [Translations]()
7. [Github Issue Labels](#github-issue-labels)
    

## Overview

> Digital Democracy's codebase is maintained through Github Version Control. Pull requests can be made if you would like to contribute to our codebase. Before creating a pull request please refer to our [contributing doc]().

Reviewer: technicality:

```suggestion
Digital Democracy's codebase is maintained using Git and hosted on GitHub. Pull requests can be made if you would like to contribute to our codebase. Before creating a pull request please refer to our [contributing doc]().
```

* Include translated strings with approriate descriptions.

    > Check that you have included the localization/translation for all strings. Make sure the string can be understood by an external translator who may not have any context. If it is ambiguous, add a description. You can learn more about translations [here]()

* Point the base branch to `develop`, unless you are working on smaller pieces of a bigger project (eg. sprints, new feature).
    > All pr's should have `develop` as the base branch. The exception is when you are creating smaller pieces of a larger feature. If that is the case, the base branch of this larger feature branch should be `develop`. You can learn more about how to change the base branch [here]()

* Take out all unnecessary comments.
    > Any comments in the code that do not communicate a meaningful message to another developer should be changed to a meaningful message or taken out of the code. GIVE EXAMPLE

* Code has been tested.
    > Make sure that you have fully tested your code on a device or emulator. It is up to you as the author, to make sure everything is functioning the way that you intended. 

* Be Timely.

>     > After someone has reviewed you PR and left comments, try to address those comments within 48 hours. This keeps it fresh for yourself and the reviewer, making it easier to understand the comments.

Reviewer: similar [comment](https://github.com/digidem/mapeo-mobile/pull/740/files#r709545097) about using hard numbers for this

    <!-- I feel like i need to explain this a little better -->

* Make sure test passes CI.
    >Our CI tests take a few minutes to run after a PR has been created in Github. Check to make sure that these tests have passed. If they did not pass, see troubleshooting (TODO).

> ### **Reviwing PR**

Reviewer 2: ```suggestion
### **Reviewing PR**
```

* Don't Fix the code, just comment on Github
    > If you see any mistakes, just add a comment in github instead of going in and trying to fix it yourself. This keeps the edit history clean and unambigous as to where the contribution came from.

* Be timely

>     > When reviewing, make sure to review the entire PR. Commenting on part of the code, while not looking at other parts of the code may lead to confusion down line. Try to address any replies from the author within 48 hours. This keeps it fresh for yourself and the contributor, making it easier to understand the comments.

Reviewer: I like the hard number here but wondering if we should be that rigid about it vs saying something like "in a timely manner" 🤷‍♂️ guess it depends on how specific we want these guidelines to be

Author: Yes! Good point, I am trying to imply that they should TRY. I can change the wording to make it seem more like a suggestion then a hard fast rule

* Be Empathetic
    > Give feedback that humanizes the author, and recognizes the work that has gone into it. Try to use gentle language. Ask questions, when you are unsure, instead of assuming that something is done incorecctly. Use "I" instead of "you" when giving feedback. For example, dont say "You did this wrong", say "I would have done this x way, can you explain why it was done this way?". 

## Template
> # Header
> Give the header a meaningful name that describes what you are doing. For Example, "New Page" is too broad of a name that doesn't specifiy what bug fix has been done. Instead use "Intro Page for App tutorial Created." Try to be descriptive, while also being concise.
>
># Description
> Describe What exactly has been done:
> Give a highlevel overview of what work has been done.
>
> If there has been a change in the UI, also describe how to view this change in the app. When necessary give step by step instructions on what steps to recreate in order to view this change.
>
> If this is a bug fix, describe what the original bug was.
>
> Add Screen shots of the new UI
>
> # Wireframes

> > Include link to wireframes here

Reviewer: i think we can generalize this to something like `Include any relevant screenshots or links` maybe with a qualifier along the lines of `e.g. screenshots of changes in UI`. I'm assuming that this is a resource that should apply to a broader scope of contributors, whether they're outside the team or contributing to the backend (could be mistaken)

Author: Good point, I want to separate screenshots and wireframes. I think people should show screenshots regradless AND if they have wireframe, then to refer to them here. So I will make it more obvious that they should include it if they have it!

>
> # To Do
> Include any tasks that still need to be completed that cannot be addressed in this PR
GIVE EXAMPLES HERE

> ## How To

Reviewer: this might be an iffy section because it'll largely vary based on the git interface they're using (e.g. GitHub's desktop app, git cli, gh cli client, etc). generally would point to official documentation about each thing if available (i'm sure GitHub has some pages dedicated to some of these)

Author: Right, so I think the majority of people will be using browser, and if not, then I think they can copy and paste it off this document. I think there is a lot of value to having a template.

Reviewer: so would you just list CLI commands in this section? My point is that you can have many variations of each step depending on the interface that's being used, which will make this document pretty verbose if you try to document all of them.
